### PR TITLE
fix(node/rolldown): mark all tsfn as weak to avoid long exit time

### LINKED ISSUE #928: Action benchmark-node hang forever
This PR seems cause action benchmark-node hang forever. https://github.com/rolldown/rolldown/actions/runs/8754292734/job/24025730323

_Originally posted by @hyf0 in https://github.com/rolldown/rolldown/issues/916#issuecomment-2067638028_

### PR DESCRIPTION
**Pull request by hyf0**
### Description
- Closes https://github.com/rolldown/rolldown/issues/928.
- Closes https://github.com/rolldown/rolldown/pull/985.

Notice that this will break the implementation of `ParallelPlugin` due to worker doesn't support creating weak tsfn https://github.com/rolldown/rolldown/pull/985#issuecomment-2081384205.

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2135** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2135?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by IWANABETHATGUY** [comment]
I think this did not solve the problem, 
```bash
<DIR>/index.js  chunk │ size: 222.05 kB                                                                                                                                                    
                                                                                                                                                                                           
✔ Finished in 153.65 ms                                                                                                                                                                   

________________________________________________________
Executed in    8.41 secs      fish           external
   usr time  581.94 millis  617.00 micros  581.33 millis
   sys time  148.63 millis  189.00 micros  148.44 millis
```
here is the result in local computer, maybe this is not the right timing to remove the hacky `process.exit`

**Comment by hyf0** [comment]
@IWANABETHATGUY Could you try it again with new commits?

**Review by IWANABETHATGUY — APPROVED** [review]

File: packages/rolldown/src/cli/index.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ async function main() {
 
   if (cliOptions.config) {
     await bundleWithConfig(cliOptions.config, cliOptions)
-    process.exit(0)
```
Comment (IWANABETHATGUY) [inline packages/rolldown/src/cli/index.ts:13]: I think we could also remove the rest of `process.exit`?

Comment (hyf0) [reply packages/rolldown/src/cli/index.ts:13]: done

**Pull request merged**